docs/mvvm/source-gen/diagnostics/MVVMTK0020.md

```diff
@@ -0,0 +1,31 @@
+---
+title: MVVM Toolkit error MVVMTK0020
+author: Sergio0694
+description: MVVM Toolkit error MVVMTK0020
+keywords: community toolkit, dotnet, csharp, mvvm, net core, net standard, source generators
+dev_langs:
+  - csharp
+---
+
+# MVVM Toolkit error MVVMTK0020
+
+Fields not annotated with `[ObservableProperty]` cannot use `[NotifyPropertyChangedFor]`, `[NotifyCanExecuteChangedFor]`, `[NotifyPropertyChangedRecipients]` and `[NotifyDataErrorInfo]`.
```

Comment:
What about `[INotifyPropertyChange]` attribute, should they work there too?

Reply:
No that's different: this list is only calling out the attributes that act as _modifiers_ for `[ObservableProperty]`, and that are specifically meant to be used together with that (and that don't do anything on their own). `[INotifyPropertyChanged]` instead can just be used on a class and it works just fine, that's how it's meant to be used. Does that make sense? 🙂

Reply:
I think I read `ObservableObject` instead of `ObservableProperty` and got confused... Think we're good.

Reply:
Ah, I think it also doesn't help that `MyViewModel` doesn't have the `ObservableObject` attribute or inherit from it, as that'd be another error on top of things too, eh?

Reply:
That would be an error if the field had `[ObservableProperty]`, yes. Since it doesn't have that in the sample, not having `ObservableObject` doesn't change anything, as its APIs wouldn't be used here anyway.